grafana/k6-studio
# feat: persist proxy preferences

## Description


This PR adds a "Settings" page where users can set some preferences. The following was implemented:

### Recorder

- browser path (let users input an alternative path to where Chrome executable is located)

### Proxy

- mode (regular or upstream)
- port
- automatically find port (let k6 studio find an available port if `port` is busy)
- upstream server and authentication

## How to Test



- Launch k6 studio and check that the settings JSON file was created in (`/Users/username/Library/Application Support/k6 Studio`)
- Change the browser path and attempt to launch the recorder
- Change the proxy port and check that a new process is running on that port (example: `lsof -i tcp:6001`)
- Check that all changes are saved in the JSON file

## Checklist



## Screenshots (if appropriate):

![image](https://github.com/user-attachments/assets/e63deda7-2294-4e5a-b388-59ac67402dd7)


## Related PR(s)/Issue(s)





 Resolves https://github.com/grafana/k6-studio/issues/236
